src/Compilers/Core/Portable/InternalUtilities/IncrementalHash.cs

```diff
@@ -27,73 +49,103 @@ public static IncrementalHash Create(AssemblyHashAlgorithm hashAlgorithm)
             }
         }
 
+        /// <summary>
+        /// CoreCLR implementation.
+        /// </summary>
         private sealed class Core : IncrementalHash
         {
             // IncrementalHash
             private readonly IDisposable _incrementalHashImpl;
 
             internal Core(AssemblyHashAlgorithm hashAlgorithm)
             {
-                var name = CryptographicHashProvider.TryGetHashAlgorithmNameObj(hashAlgorithm);
+                var name = TryGetHashAlgorithmNameObj(hashAlgorithm);
                 _incrementalHashImpl = PortableShim.IncrementalHash.CreateHash(name);
             }
 
+            /// <summary>
+            /// Returns the actual FX implementation of HashAlgorithmName for given hash algorithm id.
+            /// </summary>
+            private static object TryGetHashAlgorithmNameObj(AssemblyHashAlgorithm algorithmId)
+            {
+                switch (algorithmId)
+                {
+                    case AssemblyHashAlgorithm.Sha1:
+                        return PortableShim.HashAlgorithmName.SHA1;
+
+                    default:
+                        // More algorithms can be added as needed.
+                        return null;
```

Comment:
I am still not comfortable with returning null for all algorithms, but SHA1. What happens if other algorithm is requested? If we support only SHA1, then public API shouldn't pretend that this is not the case, by allowing to request different algorithm.